src/model/table.py

```diff
@@ -20,8 +40,49 @@ def set(self):
     def delete(self):
         pass
 
-    def get(self):
-        pass
+    def get(self, query):
+        best_search = self.__get_efficient_primary_keys__(query)
+        found_objects = self.get_rows(best_search)
+        return self.__compare_found__(found_objects, query)
 
-    def get_by_primary_key(self):
-        pass
+    def __get_efficient_primary_keys__(self, query):
+        index_names = self.__table_metadata__.get_indices_names()
+        primary_keys = []
+        if query is not None or query != "" or not str(query).isspace:
+            if self.__table_metadata__.primary_key in query.values:
+                primary_keys.append(query[self.__table_metadata__.primary_key])
+            else:
+                for item in query.items():
+                    if item[0] in index_names:
+                        index_keys = self.__table_metadata__.get_index_primary_keys(str(item[0]), str(item[1]))
+                        primary_keys = index_keys if len(primary_keys) == 0 or len(primary_keys) > len(index_keys) else primary_keys
+        if len(primary_keys) == 0:
+            for primary_key in os.listdir(self.__path__):
+                if ".json" in str(primary_key) and "schema" not in str(primary_key):
+                    primary_keys.append(self.get_by_primary_key(str(primary_key).replace(".json", '')))
+        return primary_keys
+
+    def get_by_primary_key(self, primary_key):
+        path = os.path.join(self.__path__, "{}.json".format(primary_key))
+        if not os.path.exists(path):
+            return None
+        with open(path, 'r') as file:
+            return json.load(file)
+
+    @staticmethod
+    def __compare_found__(found_objects, query):
```

Comment:
This method is not understandable, rename it better, write documentation. Try to make it O(n)